hazelcast/src/main/java/com/hazelcast/ringbuffer/impl/RingbufferContainer.java

```diff
@@ -257,22 +272,20 @@ public long add(T item) {
      *                                         deserialized
      */
     public long addAll(T[] items) {
-        long firstSequence = -1;
-        long lastSequence = -1;
+        long firstSequence = ringbuffer.tailSequence() + 1;
+        long lastSequence = ringbuffer.tailSequence() + 1;
 
-        for (int i = 0; i < items.length; i++) {
-            lastSequence = addInternal(items[i]);
-            if (i == 0) {
-                firstSequence = lastSequence;
-            }
-        }
         if (store.isEnabled() && items.length != 0) {
             try {
                 store.storeAll(firstSequence, convertToData(items));
```

Comment:
I think this also relies on a specific implementation ie. the sequence being incremental by one as seen in `ArrayRingbuffer`. However, if we had a different implementation of the `Ringbuffer` interface, which worked on a fixed number or sequences that we visit circularly (stupid example I know), then this would break. I think a better approach would have been to have a SequenceIterator exposed when we do `peek` on the ringbuffer, which we could use to get the next sequence in a loop (read only mode). Please don't change sth, this is a bigger change and it was always like that. I guess we will never have another implementation anyway :)

Reply:
Isn't it part of the Ringbuffer's contract that each added item receives a unique sequence so that no collision may happen? The javadoc at least advertises that on add and addAsync. Reference is also bringing an example of using the Ringbuffer as a unique ID generator.

Reply:
Yep, the example was a bad one. What I am trying to say here is that we are exposing the internals of the implementation, which is `current_seq + 1`. If that changes to be `current_seq + 10`, which is still unique, but with different step, this would break. This is only theoretical of course, that's why I wouldn't change it, I just used this opportunity to discuss it from a design point of view.

Reply:
Alright, got it. Then adding the other change you proposed (peekNextTailSequence, on its way) internalize this logic into the Ringbuffer and hides from the external world. It is cleaner that way anyways.